packages/flutter_test/lib/src/controller.dart

```diff
@@ -338,6 +338,17 @@ abstract class WidgetController {
   ///
   /// The `speed` is in pixels per second in the direction given by `offset`.
   ///
+  /// The `offset` represents the motion one's finger would perform on the
+  /// screen, where a positive [Offset.dy] means the finger moves down and a
+  /// negative value means the finger moves up, i.e. flinging downwards and
+  /// upwards respectively. Accordingly, positive [Offset.dx] values mean
+  /// flinging to the right and negative values flinging to the left.
+  /// The contents of a viewport usually follow that gesture, meaning that
```

Comment:
What does this mean by usually? What are the cases when it wouldn't?
Also, doesn't the AxisDirection mean the opposite in some cases? If so we should mention that too.

Reply:
@Piinks I mean that someone could also implement it differently. Someone could e.g. implement a zoom when flinging in their custom widget.

And I am not sure about `AxisDirection` - I do not think it reverses the scroll direction.

Reply:
> @Piinks I mean that someone could also implement it differently. Someone could e.g. implement a zoom when flinging in their custom widget.

True, that would definitely be an edge case, I was just pointing out the use of the word 'usually' is a bit confusing.

> And I am not sure about `AxisDirection` - I do not think it reverses the scroll direction.

We should check that and mention it! :)